homework/index.js

```diff
@@ -30,18 +30,119 @@
     return elem;
   }
 
+  // create reusable function (removeChildren),
+  // to remove children of an html element which is (the parent).
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
+  // defined root outside the local scope of the function to be available to all functions.
+  const root = document.getElementById('root');
+
+  // create left box (repository Info).
+  // make a function which creates the repo info as a list.
+  function createRepoInfo(selectedRepo, repoContainer) {
+    // create container for the info of the repo.
+    const listRepoInfo = createAndAppend('ul', repoContainer, { class: 'repo-list' });
+
+    // create repo information items
+    const li = createAndAppend('li', listRepoInfo, { text: 'Repository:  ', class: 'li' });
+
+    createAndAppend('a', li, {
+      target: '_blank',
+      href: selectedRepo.html_url,
+      text: selectedRepo.name,
+    });
+
+    createAndAppend('li', listRepoInfo, {
+      text: `Description: ${selectedRepo.description}`,
+      class: 'li',
+    });
+    createAndAppend('li', listRepoInfo, { text: `Forks: ${selectedRepo.forks}`, class: 'li' });
+    createAndAppend('li', listRepoInfo, {
+      text: `Updated: ${selectedRepo.updated_at}`,
+      class: 'li',
+    });
+  }
+
+  // create (right box) as a table to render contributor Info
+  // make a function which creates contributor info as a table.
+  function createContributor(selectedRepo, repoContainer) {
+    // initialize the url of contributor.
+    const contributorUrl = selectedRepo.contributors_url;
+
+    // fetch the url of the contributor and handle error and data of the contributor
+    fetchJSON(contributorUrl, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        // create table
+        const table = createAndAppend('table', repoContainer, { class: 'table' });
+
+        const tBody = createAndAppend('tbody', table);
+
+        // create row for each contributor
+        contributors.forEach(contributor => {
+          const row = createAndAppend('tr', tBody);
+
+          const imageCell = createAndAppend('td', row);
+          createAndAppend('img', imageCell, { src: contributor.avatar_url, class: 'img' });
```

Comment:
The HTML validator complains about missing `alt` attributes (the line number below refer to the generated HTML, not to this JavaScript file):

```
Error: An “img” element must have an “alt” attribute, except under certain conditions. For details, consult guidance on providing text alternatives for images.
    From line 16, column 2358; to line 16, column 2434
    Error: An “img” element must have an “alt” attribute, except under certain conditions. For details, consult guidance on providing text alternatives for images.
    From line 16, column 2537; to line 16, column 2613
    There were errors.
```

Reply:
Thanks! I will add it soon.